lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -137,15 +157,293 @@ fn main() -> Result<()> {
+                println!("{:#?}", rkth_revoke);
+            }
+            println!("No CFPA verification implemented");
+
+            println!("=== Image ====");
+            let image = std::fs::read(src_img)?;
+            let image_len = u32::from_le_bytes(image[0x20..0x24].try_into().unwrap());
+            let image_type = BootField::unpack(image[0x24..0x28].try_into().unwrap())?;
+
+            let load_addr = u32::from_le_bytes(image[0x34..0x38].try_into().unwrap());
+            let is_plain = image_type.img_type == EnumCatchAll::Enum(BootImageType::PlainImage);
+
+            if verbose {
+                println!("image length: {image_len:#x} ({image_len})");
+                println!("image type: {image_type:#?}");
+                println!("load address: {load_addr:#x}");
+            }
+            if is_plain && load_addr != 0 {
+                println!("⚠️  Load address is non-0 in a non-plain image");
+            } else if !is_plain && load_addr == 0 {
+                println!("⚠️  Load address is 0 in a non-plain image");
+            }
+
+            let secure_boot_enabled = matches!(
+                secure_boot_cfg.sec_boot_en,
+                SecBootStatus::SignedImage1
+                    | SecBootStatus::SignedImage2
+                    | SecBootStatus::SignedImage3
+            );
+            if verbose {
+                println!("secure boot enabled in CMPA: {secure_boot_enabled}");
+            }
+
+            println!("Checking TZM configuration");
+            match secure_boot_cfg.tzm_image_type {
+                TZMImageStatus::PresetTZM => {
+                    if matches!(image_type.tzm_preset, TzmPreset::NotPresent) {
+                        println!("    ❌ CFPA requires TZ preset, but image header says it is not present");
+                    } else {
+                        todo!("don't yet know how to decode TZ preset");
+                    }
+                }
+                TZMImageStatus::InImageHeader => {
+                    if matches!(image_type.tzm_image_type, TzmImageType::Enabled) {
+                        if matches!(image_type.tzm_preset, TzmPreset::Present) {
+                            todo!("don't yet know how to decode TZ preset");
+                        } else {
+                            println!("    ✅ TZM enabled in image header, without preset data");
+                        }
+                    } else {
+                        println!("    ✅ TZM disabled in image header");
+                    }
+                }
+                TZMImageStatus::DisableTZM => {
+                    if matches!(image_type.tzm_image_type, TzmImageType::Enabled) {
+                        println!(
+                            "    ❌ CFPA requires TZ disabled, but image header says it is enabled"
+                        );
+                    } else if matches!(image_type.tzm_preset, TzmPreset::Present) {
```

Comment:
Out of curiosity, why not `image_type.tzm_preset == TzmPreset::Present` ? It's shorter and (imo) more clear. Is there a reason these enums can't derive Eq?

Reply:
Historical artifact; fixed!